src/java.base/share/classes/java/util/IdentityHashMap.java

```diff
@@ -267,9 +267,25 @@ private void init(int initCapacity) {
      * @throws NullPointerException if the specified map is null
      */
     public IdentityHashMap(Map<? extends K, ? extends V> m) {
-        // Allow for a bit of growth
-        this((int) ((1 + m.size()) * 1.1));
-        putAll(m);
+        this(m, m.size());
```

Comment:
I actually do not know if this change shall be in this pr, or in a new issue.
But the original codes can cause IdentityHashMap to over-allocate at size == 19.
@stuart-marks what do you think about this?

Reply:
I checked this `(int) ((1 + m.size()) * 1.1)`. It is there when the codes original created there at 2007.

I don't thik it reasonable. or is there eveidence it be?

Reply:
> I don't thik it reasonable. or is there eveidence it be?

If this map is too dense, there may be a lot of hash collisions, and the lookup performance would decrease because this hashmap is linear probe than red-black tree buckets like the regular hash map is using. This was effectively trading some memory for better performance. You should run benchmarks to see how bad the lookup performance degrades after you saves memory used by the hash table.

Reply:
> > I don't thik it reasonable. or is there eveidence it be?
> 
> If this map is too dense, there may be a lot of hash collisions, and the lookup performance would decrease because this hashmap is linear probe than red-black tree buckets like the regular hash map is using. This was effectively trading some memory for better performance. You should run benchmarks to see how bad the lookup performance degrades after you saves memory used by the hash table.

1. `* 1.1` don't seem a very reasonable number for making it non-dense
2. the main reason I argue about is it does not make the resize at `put` and the `map constructor` follow a same resize way.

you create an IdentityHashMap `mapA`, put 19 key-value pairs, and build another IdentityHashMap using `mapB = new IdentityHashMap(mapA)`, and you can find mapB's table is twice larger than mapA's.
That seems weird and nonsence.
I can accept it if mapB's table's length is smaller or equal than mapA, but larger...why need to make it larger when we copy a map?

Reply:
> You should run benchmarks to see how bad the lookup performance degrades after you saves memory used by the hash table.
OK, would find some time for it.

Reply:
> > You should run benchmarks to see how bad the lookup performance degrades after you saves memory used by the hash table.
> 
> OK, would find some time for it.

@liach which jmh test should I run by the way?

Or is there some commandline tools for this?

Reply:
Unfortunately I don't think there is any for now. Maybe someone familiar with the identity hash map can comment on your changes to the expected map size

Reply:
> Unfortunately I don't think there is any for now. 

Yes. that is why I ask is there evidence to show `(int) ((1 + m.size()) * 1.1)` be an optimization, but not otherwise.
(IMO it is otherwise.)